lib/fields.c

```diff
@@ -91,12 +92,7 @@ void change_field (char *buf, size_t maxsize, const char *prompt)
 		 * entering a space.  --marekm
 		 */
 		stpcpy(strrspn(newf, " \t\n"), "");
-
-		cp = newf;
-		while (isspace (*cp)) {
-			cp++;
-		}
-
+		cp = stpspn(newf, " \t");
```

Comment:
How was the tab calculated previously? Was it the equivalent of 1, 4, 8... characters? And now?

Reply:
```diff
$ git log -p alx/master..str -- lib/fields.c
commit 75c094da708e9da091e5b0e737b704f2e816b140
Author: Alejandro Colomar <alx@kernel.org>
Date:   Wed Jul 3 22:47:45 2024 +0200

    lib/, src/: Use stpspn() instead of its pattern
    
    Signed-off-by: Alejandro Colomar <alx@kernel.org>

diff --git a/lib/fields.c b/lib/fields.c
index 555f4bed..32399fe5 100644
--- a/lib/fields.c
+++ b/lib/fields.c
@@ -16,6 +16,7 @@
 #include <stdio.h>
 
 #include "prototypes.h"
+#include "string/strchr/stpspn.h"
 #include "string/strchr/strrspn.h"
 #include "string/strtok/stpsep.h"
 
@@ -91,12 +92,7 @@ void change_field (char *buf, size_t maxsize, const char *pr>
                 * entering a space.  --marekm
                 */
                stpcpy(strrspn(newf, " \t\n"), "");
-
-               cp = newf;
-               while (isspace (*cp)) {
-                       cp++;
-               }
-
+               cp = stpspn(newf, " \t");
                strcpy (buf, cp);
        }
 }
```

The tab is one byte.  It doesn't matter how it is interpreted; as it is stored in the string as a single `0x09` byte.  isspace('\t') returned true, an cp++ advanced one byte.

Reply:
See also ascii(7).  <https://www.man7.org/linux/man-pages/man7/ascii.7.html>